nvflare/app_common/utils/fl_model_utils.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvflare.apis.dxo import DXO, DataKind, from_shareable
+from nvflare.apis.shareable import Shareable
+from nvflare.app_common.abstract.fl_model import FLModel, FLModelConst, TransferType
+from nvflare.app_common.app_constant import AppConstants
+
+MODEL_ATTRS = [
+    FLModelConst.MODEL,
+    FLModelConst.METRICS,
+    FLModelConst.TRANSFER_TYPE,
+    FLModelConst.OPTIMIZER,
+    FLModelConst.METRICS,
+    FLModelConst.CONFIGS,
+    FLModelConst.CLIENT_WEIGHTS,
+    FLModelConst.ROUND,
+    FLModelConst.TOTAL_ROUNDS,
+    FLModelConst.META,
+]
+
+
+transfer_type_to_data_kind = {
+    TransferType.MODEL.value: DataKind.WEIGHTS,
+    TransferType.MODEL_DIFF.value: DataKind.WEIGHT_DIFF,
+    TransferType.METRICS_ONLY.value: DataKind.METRICS,
+}
+data_kind_to_transfer_type = {v: k for k, v in transfer_type_to_data_kind.items()}
+
+
+class FLModelUtils:
+    @staticmethod
+    def to_shareable(fl_model: FLModel) -> Shareable:
+        """From FLModel to NVFlare side shareable.
+
+        This is a temporary solution to converts FLModel to the shareable of existing style,
+        so that we can reuse the existing components we have.
+
+        In the future, we should be using the to_dxo, from_dxo directly.
+        And all the components should be changed to accept the standard DXO.
+        """
+        if fl_model.transfer_type.value not in transfer_type_to_data_kind:
+            raise ValueError(f"Invalid TransferType: ({fl_model.transfer_type.value}).")
+
+        data_kind = transfer_type_to_data_kind[fl_model.transfer_type.value]
+
+        dxo = DXO(data_kind, data=fl_model.model, meta={})
+        shareable = dxo.to_shareable()
+        if fl_model.round:
```

Comment:
This is incorrect because fl_mode.round could be 0, which is a valid round.
You should check:
if fl_model.round is not None